Week7/CatApi/app/src/main/java/com/example/catapi/ui/SecondFragment.kt

```diff
@@ -0,0 +1,195 @@
+package com.example.catapi.ui
+
+import android.net.ConnectivityManager
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.findNavController
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.example.catapi.LoadingDialog
+import com.example.catapi.MainActivity
+import com.example.catapi.R
+import com.example.catapi.model.*
+import com.example.catapi.networking.*
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import kotlinx.android.synthetic.main.fragment_second.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.lang.reflect.Type
+
+
+/**
+ * A simple [Fragment] subclass as the second destination in the navigation.
+ */
+@RequiresApi(Build.VERSION_CODES.M)
+class SecondFragment : Fragment() {
+
+    private lateinit var breedMap : Map<String,String>
+    private var image = ""
+    private lateinit var model : MyViewModel
+
+
+    private val networkStatusChecker by lazy {
+        NetworkStatusChecker(activity?.getSystemService(ConnectivityManager::class.java))
+    }
+
+    // ------------------------------------ For Api with List<Breed> adapter
+    private val apiServiceBreed by lazy {
+        val type: Type = Types.newParameterizedType(List::class.java, Breed::class.java)
+        val parserM = Moshi.Builder().build()
+        val reportAdapter: JsonAdapter<List<Breed>> = parserM.adapter(type)
+        buildApiService(parserM)
+    }
+
+    private val remoteApiBreed by lazy {
+        RemoteApi(apiServiceBreed)
+    }
+    //----------------------------------------------------------------------
+
+    // ------------------------------------ For Api with List<Cat> adapter
+    private val apiServiceImage by lazy {
+
+        val type: Type = Types.newParameterizedType(List::class.java, Cat::class.java)
+        val parserM = Moshi.Builder().build()
+        val reportAdapter: JsonAdapter<List<Cat>> = parserM.adapter(type)
+        buildApiService(parserM)
+
+    }
+
+    private val remoteApiImage by lazy {
+        RemoteApi(apiServiceImage)
+    }
+    //----------------------------------------------------------------------
+
+    private val genreList = arrayOf("Male","Female")
+
+    private fun obtainBreedList(){
+        val loadingDialog = activity?.let { LoadingDialog(it) }
+        loadingDialog?.startDialog()
+
+        GlobalScope.launch(Dispatchers.Main) {
```

Comment:
Watch out for GlobalScope, it is better to use a different scope for coroutines.